### PR TITLE
Docker build configuration

### PR DESCRIPTION
**Pull request by dhanrajteqfocus002**

### CODE REVIEW

**Comment by SeanSCao** [comment]
Build works for me. Could we change the docker service name from sesar to sesar-rest-api just for clarity sake? Also we should be able to remove the -DskipTests at this point right?

**Comment by dhanrajteqfocus002** [comment]
Changing docker service name from sesar to sesar-rest-api in docker-compose is good idea, will change the service name.
We need to keep -DskipTests as of now because we are using github action to build docker images and we don't know what IP the request will be coming to db while building process.

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
 RUN /bin/rm -fR /usr/local/tomcat/webapps/*
-COPY target/sesar-rest-api-1.0.0.war /usr/local/tomcat/webapps/api.war
+RUN sed -i 's/8080/80/' /usr/local/tomcat/conf/server.xml
+WORKDIR /usr/local/tomcat/webapps/
+COPY --from=maven-build /usr/src/app/target/sesar-rest-api-1.0.0.war /usr/local/tomcat/webapps/api.war
 RUN sh -c 'touch /usr/local/tomcat/webapps/api.war'
 # Expose Ports. SESAR will run on 443. Any request to 80 will be redirect to 443
-EXPOSE 443 8080
+EXPOSE 443 80
 ENV JAVA_OPTS="-XX:PermSize=4096m -XX:MaxPermSize=1024m"
```
Comment (davestern) [inline Dockerfile:16]: Is this necessary? Doesn't this cause problems if our task size is smaller than this?

File: .github/workflows/build_publish.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          tags: |
+            ${{ steps.login-ecr.outputs.registry }}/${{ inputs.APPLICATION }}:${{ env.SHA }}
```
Comment (davestern) [inline .github/workflows/build_publish.yml]: newline needed

File: .github/workflows/deploy_development.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  deploy_ecs_development:
+    needs: [build_publish_development]
+    uses: ./.github/workflows/deploy_ecs.yml
+    with:
+      APPLICATION: sesar-rest-api
+      ENVIRONMENT: production
+    secrets:
+      AWS_ACCESS_KEY_ID: ${{ secrets.SESAR_AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.SESAR_AWS_SECRET_ACCESS_KEY }}
+      AWS_REGION: ${{ secrets.SESAR_AWS_REGION }}
```
Comment (davestern) [inline .github/workflows/deploy_development.yml]: newline needed

File: .github/workflows/deploy_ecs.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          task-definition: task-definition.json
+          container-name: ${{ inputs.APPLICATION }}
+          image: ${{ steps.login-ecr.outputs.registry }}/${{ inputs.APPLICATION }}:${{ env.SHA }}
+      - name: Deploy Amazon ECS task definition
+        uses: aws-actions/amazon-ecs-deploy-task-definition@v1
+        with:
+          task-definition: ${{ steps.task-def.outputs.task-definition }}
+          service: ${{ inputs.ENVIRONMENT }}-${{ inputs.APPLICATION }}
+          cluster: ${{ inputs.ENVIRONMENT }}
+          wait-for-service-stability: true
```
Comment (davestern) [inline .github/workflows/deploy_ecs.yml]: newline needed

File: Dockerfile [hunk 5]
```diff
@@ -1,9 +1,17 @@
...
+COPY . /usr/src/app
+RUN mvn package -DskipTests
+
+FROM tomcat:8.5-alpine
 # VOLUME /tmp
 RUN /bin/rm -fR /usr/local/tomcat/webapps/*
-COPY target/sesar-rest-api-1.0.0.war /usr/local/tomcat/webapps/api.war
+RUN sed -i 's/8080/80/' /usr/local/tomcat/conf/server.xml
+WORKDIR /usr/local/tomcat/webapps/
+COPY --from=maven-build /usr/src/app/target/sesar-rest-api-1.0.0.war /usr/local/tomcat/webapps/api.war
```
Comment (davestern) [inline Dockerfile:12]: Nice!

File: Dockerfile [hunk 6]
```diff
@@ -1,9 +1,17 @@
...
 RUN sh -c 'touch /usr/local/tomcat/webapps/api.war'
```
Comment (davestern) [inline Dockerfile:13]: Not urgent, but running as root is insecure. Don't change now, but we should make a note to update this and SESAR

Example:

```
FROM python:3.9-alpine
RUN addgroup -S appgroup \

&& adduser -S appuser -G appgroup


USER appuser
[REST_OF_YOUR_DOCKERFILE]
```

File: docker-compose.yml [hunk 7]
```diff
@@ -1,11 +1,11 @@
 version: '3'
 services:
-  sesar:
+  sesar-rest-api:
     image: earthchemdocker/sesar-rest-api:1.0.0
     container_name: sesar-rest-api-1.0.0
     ports:
-     - "8080:8080"
+     - "8080:80"
```
Comment (davestern) [inline docker-compose.yml]: Wondering again if we should change this or live with 8080. We did this for consistency right? Does it really help?

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -1,7 +1,7 @@
-spring.datasource.url=
-spring.datasource.username=
-spring.datasource.password=
-
+spring.datasource.url=jdbc:postgresql://${SESAR_DB_SERVER_NAME}:${SESAR_DB_PORT}/${SESAR_DB_NAME}
+spring.datasource.username=${SESAR_DB_WEB_LOGIN}
+spring.datasource.password=${SESAR_DB_WEB_PASS}
```
Comment (davestern) [inline src/main/resources/application.properties:3]: 👍

**Review by davestern — CHANGES_REQUESTED** [review]
Check the newlines and the `JAVA_OPTS` that set memory. Concerned about how that will work in Fargate

File: Dockerfile (hunk 1 above)
Comment (dhanrajteqfocus002) [reply Dockerfile:16]: Image created from this Dockerfile is already running in staging and production ECS Cluster. Not sure if this is necessary, didn't make any changes to JAVA_OPTS it is from previous commits.

File: docker-compose.yml (hunk 7 above)
Comment (dhanrajteqfocus002) [reply docker-compose.yml]: Right we should map it to 80 for consistency, changing it to port 80.

File: docker-compose.yml [hunk 9]
```diff
@@ -1,11 +1,11 @@
 version: '3'
 services:
-  sesar:
+  sesar-rest-api:
     image: earthchemdocker/sesar-rest-api:1.0.0
     container_name: sesar-rest-api-1.0.0
     ports:
-     - "8080:8080"
+     - "80:80"
```
Comment (davestern) [inline docker-compose.yml:7]: Doesn't this have to be `8080:80`? I believe the app listens on 8080 but we want to expose 80 right?

**Review by davestern — APPROVED** [review]

**Pull request merged**